apiv2/models.py

```diff
@@ -105,3 +102,20 @@ def client_secret(self):
     @property
     def version(self):
         return "V2"
+
+
+class DownloadToken(models.Model):
```

Comment:
Why not using something like JWT tokens for this? That way you don't need to store anything in the DB!

As the length of the token in this case is irrelevant you can dump all the information you need in the token JSON.

Sign it with a private key and you can check the signature when you receive the token again.